tests/Tests/Connection/Http/HttpConnectionTests.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Net.Http;
+using FluentAssertions;
+using OpenSearch.Client;
+using OpenSearch.Net;
+using OpenSearch.OpenSearch.Xunit.XunitPlumbing;
+using Xunit;
+using HttpMethod = OpenSearch.Net.HttpMethod;
+
+namespace Tests.Connection.Http;
+
+public class HttpConnectionTests
+{
+	private static readonly System.Net.Http.HttpMethod Patch = new System.Net.Http.HttpMethod("PATCH");
+
+    private class MockHttpConnection : HttpConnection
+    {
+        public HttpRequestMessage LastRequest { get; private set; }
+
+        protected override HttpRequestMessage CreateHttpRequestMessage(RequestData requestData)
+        {
+            LastRequest = base.CreateHttpRequestMessage(requestData);
+            return LastRequest;
+        }
+
+		public override TResponse Request<TResponse>(RequestData requestData)
+		{
+			CreateHttpRequestMessage(requestData);
+			return new TResponse();
+		}
+	}
+
+    public static TheoryData<HttpMethod> HttpMethods()
+    {
+        var data = new TheoryData<HttpMethod>();
+        foreach (var httpMethod in Enum.GetValues<HttpMethod>()) data.Add(httpMethod);
+		return data;
+    }
+
+    [TU]
+    [MemberData(nameof(HttpMethods))]
+    public void UsesCorrectHttpMethod(HttpMethod method)
+    {
+        var mockHttpConnection = new MockHttpConnection();
+        mockHttpConnection.Request<StringResponse>(new RequestData(method, "", null, null, null, null));
+		mockHttpConnection.LastRequest.Method.Should().Be(ConvertHttpMethod(method));
+    }
+
+    private static System.Net.Http.HttpMethod ConvertHttpMethod(HttpMethod httpMethod) =>
+        httpMethod switch
+        {
+            HttpMethod.GET => System.Net.Http.HttpMethod.Get,
+            HttpMethod.POST => System.Net.Http.HttpMethod.Post,
+            HttpMethod.PUT => System.Net.Http.HttpMethod.Put,
+            HttpMethod.DELETE => System.Net.Http.HttpMethod.Delete,
+            HttpMethod.HEAD => System.Net.Http.HttpMethod.Head,
+            HttpMethod.PATCH => Patch,
+            _ => throw new ArgumentException("Invalid value for HttpMethod", nameof(httpMethod))
+        };
+
```

Comment:
Rather than duplicating the switch statement here, it'd be better to simplify the test side of this to something like:
```suggestion
		mockHttpConnection.LastRequest.Method.Should().Be(new System.Net.Http.HttpMethod(method.ToString()));
    }
```